plugin/trino-mariadb/src/test/java/io/trino/plugin/mariadb/TestMariaDbTypeMapping.java

```diff
@@ -812,6 +813,76 @@ private void testTimestamp(ZoneId sessionZone)
                 .execute(getQueryRunner(), session, trinoCreateAndInsert("test_timestamp"));
     }
 
+    @Test
+    public void testDatetime()
+    {
+        testDatetime(UTC);
+        testDatetime(jvmZone);
+        testDatetime(vilnius);
+        testDatetime(kathmandu);
+        testDatetime(TestingSession.DEFAULT_TIME_ZONE_KEY.getZoneId());
+    }
+
+    private void testDatetime(ZoneId sessionZone)
+    {
+        Session session = Session.builder(getSession())
+                .setTimeZoneKey(getTimeZoneKey(sessionZone.getId()))
+                .build();
+
+        SqlDataTypeTest.create()
+                .addRoundTrip("datetime(3)", "'1958-01-01 13:18:03.123'", createTimestampType(3), "TIMESTAMP '1958-01-01 13:18:03.123'")
```

Comment:
Please update `testTimestamp(ZoneId sessionZone)` method to accept both timestamp and datetime.

Reply:
`MariaDB stores values that use the DATETIME data type in a format that supports values between 1000-01-01 00:00:00.000000 and 9999-12-31 23:59:59.999999` which is different from TIMESTAMP (https://mariadb.com/kb/en/timestamp/#supported-values).  As the test cases has different ranges of values, so it is better to keep in separate.

Reply:
Please leave the code comment.

Reply:
Yea, adding a common roudrips method for datetime and timestamp test.